apps/condo/domains/user/schema/User.test.js

```diff
@@ -455,6 +459,89 @@ describe('User fields', () => {
     })
+
+    describe('name', () => {
+        describe('Inside list queries (allUsers, User)', () => {
+            test('Can be seen only by admin / support / direct accessed users', async () => {
+                const admin = await makeLoggedInAdminClient()
+                const support = await makeClientWithSupportUser()
+
+                const [rightsSet] = await createTestUserRightsSet(admin, {
+                    canReadUsers: true,
+                })
+                const directAccessedClient = await makeClientWithNewRegisteredAndLoggedInUser({
+                    rightsSet: { connect: { id: rightsSet.id } },
+                })
+
+                const defaultClient = await makeClientWithNewRegisteredAndLoggedInUser()
+
+                expect(admin.user.name).not.toBeNull()
+                expect(support.user.name).not.toBeNull()
+                expect(directAccessedClient.user.name).not.toBeNull()
+                expect(defaultClient.user.name).not.toBeNull()
+
+                const where = {
+                    id_in: [support.user.id, directAccessedClient.user.id, defaultClient.user.id],
+                }
+
+                for (const client of [admin, support, directAccessedClient]) {
+                    const users = await User.getAll(client, where)
+                    expect(users).toHaveLength(3)
+                    expect(users).toEqual(expect.arrayContaining([
+                        expect.objectContaining({ id: support.user.id, name: support.user.name }),
+                        expect.objectContaining({ id: directAccessedClient.user.id, name: directAccessedClient.user.name }),
+                        expect.objectContaining({ id: defaultClient.user.id, name: defaultClient.user.name }),
+                    ]))
+
+                    const user = await User.getOne(client, { id: defaultClient.user.id })
+                    expect(user).toEqual(expect.objectContaining({
+                        name: defaultClient.user.name,
+                    }))
+                }
+
+                const { data, errors } = await User.getAll(defaultClient, where, { raw: true })
+                expect(data).toHaveProperty('objs')
+                expect(data.objs).toHaveLength(3)
+                expect(data.objs).toEqual(expect.arrayContaining([
+                    expect.objectContaining({ id: support.user.id, name: null }),
+                    expect.objectContaining({ id: directAccessedClient.user.id, name: null }),
+                    expect.objectContaining({ id: defaultClient.user.id, name: defaultClient.user.name }),
+                ]))
+
+                const nameAccessDeniedErrors = errors.filter(error =>
+                    error && error.name &&
+                    error.path && error.name === 'AccessDeniedError' &&
+                    error.path.length === 3 &&
+                    error.path[0] === 'objs' && typeof error.path[1] === 'number' && error.path[2] === 'name'
+                )
```

Comment:
That's really hard to understand)

Reply:
You're trying to collect name related error from all of the sources, right?

Reply:
Yep, not the best try-catch here, but we don't have utils for that :(

I need to check partial data (only self-name returned) and partial errors (only with path ['objs', idx, 'name']) since there's extra errors about right set

Reply:
Yup, thats true)